src/audio/copier/copier.c

```diff
@@ -1313,7 +1315,30 @@ static int copier_params(struct comp_dev *dev, struct sof_ipc_stream_params *par
 				ret = cd->endpoint[i]->drv->ops.params(cd->endpoint[i],
 								       &demuxed_params);
 			} else {
-				ret = cd->endpoint[i]->drv->ops.params(cd->endpoint[i], params);
+				if (dev->ipc_config.type == SOF_COMP_HOST) {
+					child_dev = cd->endpoint[i];
+					ret = host_verify_params(child_dev, params);
+					if (ret < 0) {
+						comp_err(dev, "host_params(): pcm params verification failed.");
+						return -EINVAL;
+					}
+
+					is_scheduling_source =
+						child_dev == child_dev->pipeline->sched_comp;
+					/* copier dev only used for log print */
+					ret = host_params_dma(cd->hd, dev, params,
+							      &child_dev->bsink_list,
+							      &child_dev->bsource_list,
+							      child_dev->pipeline,
+							      child_dev->frames,
+							      is_scheduling_source);
+					/* set up callback */
```

Comment:
this part still use host dev, @ranj063 , I think we need trying to remove it as well, do you think so?

Reply:
Yes, certainly child_dev->bsink_list and bsource_list will be pointing to the endpoint buffers that we eventually want to get rid of